src/mqt/predictor/ml/predictor.py

```
@@ -356,15 +419,289 @@ def _generate_training_sample(
+        self,
+        trial: optuna.Trial,
+        dataset: NDArray[np.float64] | list[torch_geometric.data.Data],
+        task: str,
+        in_feats: int,
+        num_outputs: int,
+        loss_fn: nn.Module,
+        k_folds: int,
+        batch_size: int = 32,
+        num_epochs: int = 10,
+        patience: int = 10,
+        verbose: bool = False,
```

Ruff (0.14.8)

Boolean-typed positional argument in function definition

(FBT001)